format/postgres/flavours/pgpro11/pg_control.go

```diff
@@ -0,0 +1,220 @@
+/*   16      |     4 */ // DBState state;
+/* XXX  4-byte hole */
+/*   24      |     8 */ // pg_time_t time;
+/*   32      |     8 */ // XLogRecPtr checkPoint;
+/*   40      |    80 */ // CheckPoint checkPointCopy;
+/*  120      |     8 */ // XLogRecPtr unloggedLSN;
+/*  128      |     8 */ // XLogRecPtr minRecoveryPoint;
+/*  136      |     4 */ // TimeLineID minRecoveryPointTLI;
+/* XXX  4-byte hole */
+/*  144      |     8 */ // XLogRecPtr backupStartPoint;
+/*  152      |     8 */ // XLogRecPtr backupEndPoint;
+/*  160      |     1 */ // _Bool backupEndRequired;
+/* XXX  3-byte hole */
+/*  164      |     4 */ // int wal_level;
+/*  168      |     1 */ // _Bool wal_log_hints;
+/* XXX  3-byte hole */
+/*  172      |     4 */ // int MaxConnections;
+/*  176      |     4 */ // int max_worker_processes;
+/*  180      |     4 */ // int max_prepared_xacts;
+/*  184      |     4 */ // int max_locks_per_xact;
+/*  188      |     1 */ // _Bool track_commit_timestamp;
+/* XXX  3-byte hole */
+/*  192      |     4 */ // uint32 maxAlign;
+/* XXX  4-byte hole */
+/*  200      |     8 */ // double floatFormat;
+/*  208      |     4 */ // uint32 blcksz;
+/*  212      |     4 */ // uint32 relseg_size;
+/*  216      |     4 */ // uint32 xlog_blcksz;
+/*  220      |     4 */ // uint32 xlog_seg_size;
+/*  224      |     4 */ // uint32 nameDataLen;
+/*  228      |     4 */ // uint32 indexMaxKeys;
+/*  232      |     4 */ // uint32 toast_max_chunk_size;
+/*  236      |     4 */ // uint32 loblksize;
+/*  240      |     1 */ // _Bool float4ByVal;
+/*  241      |     1 */ // _Bool float8ByVal;
+/* XXX  2-byte hole */
+/*  244      |     4 */ // uint32 data_checksum_version;
+/*  248      |    32 */ // char mock_authentication_nonce[32];
+/*  280      |     4 */ // pg_icu_version icu_version;
+/*  284      |     4 */ // pg_crc32c crc
+//
+/* total size (bytes):  288 */
+
+// type = struct CheckPoint {
+/*    0      |     8 */ // XLogRecPtr redo;
+/*    8      |     4 */ // TimeLineID ThisTimeLineID;
+/*   12      |     4 */ // TimeLineID PrevTimeLineID;
+/*   16      |     1 */ // _Bool fullPageWrites;
+/* XXX  3-byte hole */
+/*   20      |     4 */ // uint32 nextXidEpoch;
+/*   24      |     4 */ // TransactionId nextXid;
+/*   28      |     4 */ // Oid nextOid;
+/*   32      |     4 */ // MultiXactId nextMulti;
+/*   36      |     4 */ // MultiXactOffset nextMultiOffset;
+/*   40      |     4 */ // TransactionId oldestXid;
+/*   44      |     4 */ // Oid oldestXidDB;
+/*   48      |     4 */ // MultiXactId oldestMulti;
+/*   52      |     4 */ // Oid oldestMultiDB;
+/*   56      |     8 */ // pg_time_t time;
+/*   64      |     4 */ // TransactionId oldestCommitTsXid;
+/*   68      |     4 */ // TransactionId newestCommitTsXid;
+/*   72      |     4 */ // TransactionId oldestActiveXid;
+/* XXX  4-byte padding */
+//
+/* total size (bytes):   80 */
+
+func DecodePgControl(d *decode.D, in any) any {
+	d.SeekAbs(0)
```

Comment:
Possible to seekback in `probeForDecode` instead of in each `DecodePgControl` doing it?

Reply:
Yes, sure